src/Maths/Silk.NET.Maths/Vector2.cs

```diff
@@ -0,0 +1,551 @@
+        public static T DistanceSquared(Vector2<T> value1, Vector2<T> value2)
+        {
+            var difference = value1 - value2;
+            return Dot(difference, difference);
+        }
+
+        /// <summary>Divides the first vector by the second.</summary>
+        /// <param name="left">The first vector.</param>
+        /// <param name="right">The second vector.</param>
+        /// <returns>The vector resulting from the division.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector2<T> Divide(Vector2<T> left, Vector2<T> right)
+            => new(Scalar.Divide(left.X, right.X), Scalar.Divide(left.Y, right.Y));
+
+        /// <summary>Divides the specified vector by a specified scalar value.</summary>
+        /// <param name="left">The vector.</param>
+        /// <param name="divisor">The scalar value.</param>
+        /// <returns>The vector that results from the division.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector2<T> Divide(Vector2<T> left, T divisor)
+            => new(Scalar.Divide(left.X, divisor), Scalar.Divide(left.Y, divisor));
+
+        /// <summary>Returns the dot product of two vectors.</summary>
+        /// <param name="value1">The first vector.</param>
+        /// <param name="value2">The second vector.</param>
+        /// <returns>The dot product.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static T Dot(Vector2<T> value1, Vector2<T> value2)
+            => Scalar.Add(Scalar.Multiply(value1.X, value2.X), Scalar.Multiply(value1.Y, value2.Y));
+
+        /// <summary>Returns a boolean indicating whether the given Vector2 is equal to this Vector2 instance.</summary>
+        /// <param name="other">The Vector2 to compare this instance to.</param>
+        /// <returns>True if the other Vector2 is equal to this instance; False otherwise.</returns>
+        public readonly bool Equals(Vector2<T> other) => this == other;
+        
+        /// <summary>Returns a boolean indicating whether the given Object is equal to this Vector2 instance.</summary>
+        /// <param name="obj">The Object to compare against.</param>
+        /// <returns>True if the Object is equal to this Vector2; False otherwise.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public override readonly bool Equals(object? obj) => (obj is Vector2<T> other) && Equals(other);
+        
+        /// <summary>Returns the hash code for this instance.</summary>
+        /// <returns>The hash code.</returns>
+        public override readonly int GetHashCode() => HashCode.Combine(X, Y);
+
+        /// <summary>Returns the length of the vector.</summary>
+        /// <returns>The vector's length.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public readonly T Length() => Scalar.Sqrt(LengthSquared());
```

Comment:
Convert all propertiable methods to properties now that serializers are second class

Not urgent

Reply:
See comment about LengthSquared, this one is even more expensive.